blockchain/src/genesis.rs

```diff
@@ -44,13 +45,13 @@ pub fn genesis(keychains: &[KeyChain], amount: i64) -> Vec<Block> {
         let previous = Hash::digest(&"genesis".to_string());
         let base = BaseBlockHeader::new(version, previous, epoch, timestamp);
 
-        let witnesses = keychains
+        let witnesses: BTreeSet<cosi_keys::PublicKey> = keychains
```

Comment:
and this line could be minified 
```suggestion
        let witnesses: BTreeSet<_> = keychains
```